staging/src/k8s.io/client-go/transport/round_trippers.go

```diff
@@ -87,6 +93,17 @@ type authProxyRoundTripper struct {
 	rt http.RoundTripper
 }
 
+// NewCacheRoundTripper creates a roundtripper that reads the ETag on
+// response headers and send the If-None-Match header on subsequent
+// corresponding requests.
+func NewCacheRoundTripper(cacheDir string, rt http.RoundTripper) http.RoundTripper {
+	d := diskv.New(diskv.Options{BasePath: cacheDir})
+	t := httpcache.NewTransport(diskcache.NewWithDiskv(d))
```

Comment:
Thanks for providing more context!

I don't think you should rely on that behavior, as it would heavily depend on what cache settings the server is using. If `cache-control` starts being set, then your assumption might break.